### ISSUE TITLE
Master issue for merging old PRs

### ISSUE THREAD

**Issue by cody271** [libui-ng/libui-ng#8]

**Comment by cody271** [comment]
@szanni Added you to the table above for all of your old PRs, now that you're here. 👍 

Hope that's okay.

**Comment by szanni** [comment]
@cody271 Yes, thank you! I will be reviewing and resubmitting all my PRs over the coming days.

Should I be resubmitting issues from other people as well? Or is referencing the old issue sufficient, as some of the PRs should resolve those issues.

**Comment by cody271** [comment]
@szanni Good question actually. I leave that up to your best judgement, to be decided on an issue by issue basis. While it sure would be nice to address each and every old issue one by one, the old PRs are the priority.

**Comment by cody271** [comment]
Current plan is: before moving forward with anything new, every single one of these 82 old PRs needs to be addressed *ASAP*. I think the majority of them are probably ready to merge with trivial changes.

**Comment by cody271** [comment]
So that we didn't have to wait for each of the original authors to come back, I've actually been using the GitHub API to generate this table above and mechanically move the old PRs over to the new repo.

**Comment by rubyFeedback** [comment]
I think focusing just on merging old patches alone may already be super-helpful for people to want to aggregate. Once they see movement and momentum I am pretty confident that more and more people come and assist, if they see this fork being an opportunity.

**Comment by cody271** [comment]
> I think focusing just on merging old patches alone may already be super-helpful for people to want to aggregate. Once they see movement and momentum I am pretty confident that more and more people come and assist, if they see this fork being an opportunity.

That is the hope yes. 🙏

**Comment by DonMartio** [comment]
It is great to see so much movement. I was looking into libui#546 but i'm not shure yet if using g_ascii_strcasecmp is the best solution here. Furthermore i was starting to check for some of this warnings
`suggest braces around empty body in an ‘if’ statement`
last week.

**Comment by cody271** [comment]
@DonMartio Yeah, `g_ascii_strncasecmp()` is fine for now.

Currently `uiprivStricmp()` is only used [here](https://github.com/libui-ng/libui-ng/blob/master/common/attribute.c#L238). Though take a look at the [Windows implementation](https://github.com/libui-ng/libui-ng/blob/master/windows/text.cpp#L109) for the best way: it converts to UTF-16 and does Unicode aware comparison.

**Comment by szanni** [comment]
> [libui#484](https://github.com/andlabs/libui/pull/484) 	white_large_square 	WIP @szanni 		Fix #416, uiTableModelRowInserted on windows adds 2 rows instead of 1.

Can be updated to a check mark and:
Replaced by [libui#515](https://github.com/andlabs/libui/pull/515)

**Comment by cody271** [comment]
> > [libui#484](https://github.com/andlabs/libui/pull/484) 	white_large_square 	WIP @szanni 		Fix #416, uiTableModelRowInserted on windows adds 2 rows instead of 1.
> 
> Can be updated to a check mark and: Replaced by [libui#515](https://github.com/andlabs/libui/pull/515)

Done.

**Comment by cody271** [comment]
@szanni What else here should we label as `Out of Scope` ?

**Comment by szanni** [comment]
> [libui#273](https://github.com/andlabs/libui/pull/273) 	white_large_square 			Put window position on top-right like default Gtk apps

Honestly who cares where the tester launches? We might want an API for window placement, but I don't see any point in this.

> [libui#271](https://github.com/andlabs/libui/pull/271) 	white_large_square 			Better default position for window on macOS

Same reasoning as above.

> [libui#122](https://github.com/andlabs/libui/pull/122) 	white_large_square 			Generate OS X app bundles

Similar reasoning. Why bundle the tester? As andlabs commented a resource generator might be interesting for application bundling sometime in the distant future - but then - where do we stop? Should we not provide an nsis installer for windows? Flatpak for linux? I would personally not bother until the lib reaches _at least_ v1.

> [libui#87](https://github.com/andlabs/libui/pull/87) 	white_large_square 			Add git config.

The line feed behavior might be interesting (I don't develop on windows, so I have no idea if this is interesting to windows folks). Otherwise this just looks like a code dump to me.

**Comment by simplexidev** [comment]
> > [libui#273](https://github.com/andlabs/libui/pull/273) 	white_large_square 			Put window position on top-right like default Gtk apps
> 
> Honestly who cares where the tester launches? We might want an API for window placement, but I don't see any point in this.
> 
> > [libui#271](https://github.com/andlabs/libui/pull/271) 	white_large_square 			Better default position for window on macOS
> 
> Same reasoning as above.

I +1 the idea of adding an API for window placement.

> > [libui#87](https://github.com/andlabs/libui/pull/87) 	white_large_square 			Add git config.
> 
> The line feed behavior might be interesting (I don't develop on windows, so I have no idea if this is interesting to windows folks). Otherwise this just looks like a code dump to me.

The `.gitconfig` would be a good addition, as it will keep Visual Studio from auto changing the line endings to CRLF on Windows

**Comment by msink** [comment]
> I +1 the idea of adding an API for window placement.

Something like #62 ?

**Comment by szanni** [comment]
> [libui#546](https://github.com/andlabs/libui/pull/546) 	white_large_square 			fix warning: implicit declaration of function ‘strcasecmp’

Can be marked as done.

**Comment by simplexidev** [comment]
> > I +1 the idea of adding an API for window placement.
> 
> Something like #62 ?

Yes, but rather than the size of the window it will be the location of the top-left corner of the window on the presenting screen

**Comment by szanni** [comment]
> [libui#366](https://github.com/andlabs/libui/pull/366) 	white_large_square 			README: add project description at 

Can be marked as done. We have an about section at the top, making that PR obsolete.

**Comment by szanni** [comment]
> [libui#353](https://github.com/andlabs/libui/pull/353) 	white_large_square 			Add uiBoxChild

I personally would mark this as WONTFIX. As I commented in the original PR, the behavior can be achieved with the existing API via `uiControlParent(child) == uiControl(box)` (tested), which happens to be more efficient than the proposed new API call too. No need to clutter the API IMO.

**Comment by cody271** [comment]
> > [libui#353](https://github.com/andlabs/libui/pull/353) 	white_large_square 			Add uiBoxChild
> 
> I personally would mark this as WONTFIX. As I commented in the original PR, the behavior can be achieved with the existing API via `uiControlParent(child) == uiControl(box)` (tested), which happens to be more efficient than the proposed new API call too. No need to clutter the API IMO.

Agreed, done.

**Comment by szanni** [comment]
> libui#27

Replaced by #155

**Comment by cody271** [comment]
> > libui#27
> 
> Replaced by #155

Done.
